python/cudf/cudf/tests/test_multiindex.py

```diff
@@ -2179,3 +2179,10 @@ def test_unique_level():
     result = pd_mi.unique(level=1)
     expected = cudf_mi.unique(level=1)
     assert_eq(result, expected)
+
+
+def test_from_arrays_infer_names():
+    arrays = [pd.Index([1], name="foo"), pd.Index([2], name="bar")]
```

Comment:
Do you mind adding some more test cases for different index types? I believe I saw this bug for `pd.CategoricalIndex` too.

Reply:
Sure thing. Added more index classes here